packages/table/src/CellDefinitions/ActionCell.tsx

```diff
@@ -35,12 +105,25 @@ const ActionCell = <T extends IdType>({
           ))}
         </TableActionMenu>
         {primaryAction && isPrimaryActionVisible && (
-          <Icon
-            data-testid={`table_row_action_menu_item_${id}_primaryAction`}
-            name={primaryAction.iconName}
-            title={primaryAction.title}
-            onClick={() => primaryAction.onClick(original)}
-          />
+          <>
+            <Icon
+              id={`table_row_action_menu_item_${id}_primaryAction`}
+              data-testid={`table_row_action_menu_item_${id}_primaryAction`}
+              name={
+                typeof primaryAction.iconName === 'function' ? primaryAction.iconName(original) : primaryAction.iconName
+              }
+              onClick={() => primaryAction.onClick(original)}
+            />
+            <UncontrolledTooltip
```

Comment:
You will also see that we are updating to use Bootstrap Tooltips here instead of basic titles per design request/requirement.